runtime/config-deployer-service/ballerina/APIClient.bal

```diff
@@ -809,33 +821,53 @@ public class APIClient {
             operationPoliciesToUse = operation.operationPolicies;
         }
         if operationPoliciesToUse is APIOperationPolicies {
-            APKOperationPolicy[]? request = operationPoliciesToUse.request;
+            APKOperationPolicy[]? requestPolicies = operationPoliciesToUse.request;
+            APKOperationPolicy[]? responsePolicies = operationPoliciesToUse.response;
+            if requestPolicies is APKOperationPolicy[] && requestPolicies.length() > 0 {
+                model:HTTPRouteFilter headerModifierFilter = {'type: "RequestHeaderModifier"};
+                headerModifierFilter.requestHeaderModifier = self.extractHttpHeaderFilterData(requestPolicies, organization);
+                routeFilters.push(headerModifierFilter);
+            }
+            if responsePolicies is APKOperationPolicy[] && responsePolicies.length() > 0 {
+                model:HTTPRouteFilter headerModifierFilter = {'type: "ResponseHeaderModifier"};
+                headerModifierFilter.responseHeaderModifier = self.extractHttpHeaderFilterData(responsePolicies, organization);
+                routeFilters.push(headerModifierFilter);
+            }
         }
         return routeFilters;
     }
 
     isolated function extractHttpHeaderFilterData(APKOperationPolicy[] operationPolicy, commons:Organization organization) returns model:HTTPHeaderFilter {
+        model:HTTPHeader[] addPolicies = [];
         model:HTTPHeader[] setPolicies = [];
         string[] removePolicies = [];
         foreach APKOperationPolicy policy in operationPolicy {
-            string policyName = policy.policyName;
-
-            record {}? policyParameters = policy.parameters;
-            if (policyParameters is record {}) {
-                if (policyName == "addHeader") {
-                    model:HTTPHeader httpHeader = {
-                        name: <string>policyParameters.get("headerName"),
-                        value: <string>policyParameters.get("headerValue")
-                    };
-                    setPolicies.push(httpHeader);
+            if policy is HeaderModifierPolicy {
+                HeaderModifierPolicyParameters policyParameters = policy.parameters;
+                if policy.policyName == "AddHeaders" {
+                    ModifierHeader[] addHeaders = <ModifierHeader[]>policyParameters.headers;
+                    foreach ModifierHeader header in addHeaders {
+                        addPolicies.push(header);
+                    }
+                }
+                if policy.policyName == "SetHeaders" {
```

Comment:
We may use `enum` for the policyName.

Reply:
Fixed in https://github.com/wso2/apk/commit/d0e33c3c0dcbc8a915ee237ff0900a26148978d5